context/app/static/js/components/detailPage/files/DataProducts/hooks.ts

```diff
@@ -0,0 +1,47 @@
+import { useAppContext, useFlaskDataContext } from 'js/components/Contexts';
+import { useDetailContext } from 'js/components/detailPage/DetailContext';
+import { UnprocessedFile } from '../types';
+
+type PipelineInfo = {
+  origin: string;
+  name: string;
+};
+
+// Default fallback values in case no pipeline info is available in metadata
+const defaultPipeline: PipelineInfo = {
+  name: 'pipeline.cwl',
+  origin: 'https://github.com/hubmapconsortium/ingest-pipeline.git',
+};
+
+/**
+ * Extract the latest origin and name from the entity data in the flask context
+ * @returns {Object} { origin (pipeline URL), name (pipeline name, if specified)}
+ */
+function usePipelineInfo(): PipelineInfo {
+  const { entity } = useFlaskDataContext();
+  const dagList = entity.metadata.dag_provenance_list ?? [];
+  // Iterate over the list of DAGs and extract the latest origin and name
+  const pipelineInfo = dagList.reduce<PipelineInfo>((acc, dag) => {
+    if ('origin' in dag) {
+      acc.origin = dag.origin;
+    }
+    if ('name' in dag) {
+      acc.name = dag.name;
+    }
+    return acc;
+  }, defaultPipeline);
+  return pipelineInfo;
+}
```

Comment:
Could you explain the logic behind this?

Reply:
Sure thing!

When I was working with the dag provenance info, I noticed that some of the list items only had a pipeline name or a pipeline link, instead of having both, something like this:
```
[
  { 
     name: 'pipeline name',
  },
  {
     origin: 'https://github.com/link/to/pipeline'
  }, 

  { 
     name: 'more specific pipeline name',
  },
  {
     origin: 'https://github.com/link/to/pipeline-two'
  }, 
]
```

For example, see `metadata.dag_provenance_list` here: https://portal.test.hubmapconsortium.org/browse/dataset/277152f17b5a2f308820ab4d85c5a426.json

I put together this `.reduce` function to handle cases like this.